hackathon_site/hardware/serializers.py

```diff
@@ -45,10 +45,28 @@ class Meta:
         fields = ("id", "hardware", "order", "part_returned_health")
 
 
-class IncidentSerializer(serializers.ModelSerializer):
+class IncidentCreateSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Incident
+        fields = (
+            "id",
+            "state",
+            "time_occurred",
+            "description",
+            "order_item",
+            "created_at",
+            "updated_at",
+        )
+        read_only_fields = ("team_id", "created_at", "updated_at")
 
-    order_item = OrderItemSerializer()
+    @staticmethod
+    def get_team_id(obj: Incident):
```

Comment:
This isn't being used by `IncidentCreateSerializer`. Likewise, having `team_id` in `read_only_fields` isn't doing anything.

In order to use this, you need `team_id = serializers.SerializerMethodField()` like is on the list serializer.

Reply:
I think I commented wrong on my last review saying which of the create serializers to delete, the second one was actually the one to keep, sorry!. The definitions for both should be:

```python
class IncidentCreateSerializer(serializers.ModelSerializer):
    team_id = serializers.SerializerMethodField()

    class Meta:
        model = Incident
        fields = (
            "id",
            "state",
            "time_occurred",
            "description",
            "order_item",
            "team_id",
            "created_at",
            "updated_at",
        )
        read_only_fields = ("created_at", "updated_at")

    @staticmethod
    def get_team_id(obj: Incident) -> int:
        return obj.order_item.order.team.id


class IncidentListSerializer(IncidentCreateSerializer):
    order_item = OrderItemSerializer()
```

The only difference is that the list serializer uses a nested serializer for the order item. Everything else can stay the same.